src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala

```diff
@@ -121,10 +131,44 @@ trait ContextExtractor {
     sorted.last.tokenInterval.start - sorted.head.tokenInterval.end
   }
 
+  def getDistance(mention: Mention, lemmaIdx: (String, Int)): Int = {
+    val lemmaTokInt = Interval(lemmaIdx._2, lemmaIdx._2 + 1)
+    val sorted = Seq(mention.tokenInterval, lemmaTokInt).sorted
+    val dist = sorted.last.start - sorted.head.end
+    dist
+  }
+
   def findClosest(mention: Mention, mentions: Seq[Mention]): Mention = {
     mentions.map(m => (m, getDistance(mention, m))).minBy(_._2)._1
   }
 
-}
+  def findClosestProcessLemma(mention: Mention): String = {
+    val lemmasWithIndices = mention.sentenceObj.lemmas.get.zipWithIndex.filter(li => lemmaToProcess.contains(li._1))
+    val lemmasWithDistances = lemmasWithIndices.map(tup => (tup._1, getDistance(mention, tup)))
+      if (lemmasWithDistances.nonEmpty) {
+        lemmasWithDistances.minBy(_._2)._1
+      } else NA
```

Comment:
This didn't seem to work. Each item in the lemmasWithDistances is is a (string, int) tuple, so you can't do minBy on the item?.. Maybe I'm misunderstanding what lemmasWithDistancesOpt is supposed to be?

Reply:
This is my mistake.  I was thinking that lemmasWithDistances was an Option rather than a possibly empty collection.  Newer versions of Scala have minByOption that would be nice to (be able to) use.